cmd/starbridge/model/chain.go

```diff
@@ -26,10 +42,26 @@ func (c *Chain) String() string {
 }
 
 // NextNonce
-func (c *Chain) NextNonce() uint64 {
-	if c != ChainStellar {
-		panic(fmt.Errorf("unsupported chain %s", c.Name))
+func (c *Chain) NextNonce(sourceAccount string) (uint64, error) {
+	return c.nextNonceFn(sourceAccount)
+}
+
+func nextStellarNonceFn(sourceAccount string) (uint64, error) {
```

Comment:
I think we should attach this function to a type, a type we can instantiate in main to give it dependencies like its logger, and a horizonclient.

Reply:
yeah, that's the plan :)